PostPairsTrading.md

```diff
@@ -0,0 +1,260 @@
+
+Whether taking a technical or quantitative approach, these insights will provide valuable foundations for constructing this algorithm effectively.
+
+## In the untamed realm of the market.
+
+The market has often been described as a **stochastic** (a term which essentially means random) **process** where prices fluctuate irregularly. However, amidst this apparent randomness, we observe that **certain assets move in tandem** due to their inherent relationships. 
+
+For instance, it's logical to expect that if the prices of petrol rise, the prices of cars should also rise. This is because auto companies rely on petrol for their operations, indicating an interconnectedness between the two. Over the long run, they tend to follow similar trends, **reflecting their underlying relationship**.
+
+Is this described mathematically? **Yes**:
+
+The concept we're referring to is **cointegration** (although there are other methods, we'll focus on this one).
+
+> 💡 Which should not be confused with correlation; cointegration is a statistical property of two time series, indicating a long-term relationship between them despite short-term fluctuations. Cointegrated series move together over time, sharing a common stochastic drift. On the other hand, correlation measures the strength and direction of the linear relationship between two variables at a specific point in time. While correlation captures the degree of association between variables, cointegration reflects a deeper, underlying relationship that persists over time.
+
+Hence, we're interested in **cointegrated assets**, which are assets that exhibit the following characteristics:
+
+1. They have a similar trend, meaning the difference between both assets maintains a constant mean, and this difference fluctuates around that same mean.
+
+2. This inherent relationship persists in the long run, meaning that our series is not dependent on time.
+
+## A pair in the hand is worth two in the bush.
+
+### ADF testing
+
+Imagine **we selected 13 world indexes** and aimed to assess whether they are **cointegrated or not**. In this scenario, a crucial tool at our disposal is the Augmented Dickey-Fuller (ADF) test.
+
+> 💡 Please note that for simplicity in this code, we will be using pykx. This is necessary as we need to import our ADF test function and plot a heatmap of our results.
+ 
+> `system "l pykx.q"`
+
+Next, we import the statsmodels.tsa.stattools library and define a custom cointegration function that returns a dictionary with the ADF test results given 2 assets. **We then apply our cointegration function to the Cartesian product of every pair of assets** to get a matrix with every p-value. 
+
+![cointegration](https://github.com/hablapps/pairstrading/blob/5-Post/CointAlGif.gif?raw=true)
+
+---
+
+```q
+// Recieves 2 symbols and returns a dictionary
+
+fCoint {[assetX;assetY;]

```

Comment:
Show more q code. Maybe show how you can make use of Python code using PyKx

Reply:
Done, Altough I did not add examples of pykx use (just the line of code where we import our `coint` function). So I will not resolve this conversation because maybe you want me to add some more examples